packages/plugin-ext/src/hosted/browser/plugin-worker.ts

```diff
@@ -16,26 +16,41 @@
 import { injectable } from '@theia/core/shared/inversify';
 import { Emitter } from '@theia/core/lib/common/event';
 import { RPCProtocol, RPCProtocolImpl } from '../../common/rpc-protocol';
+import { environment } from '@theia/core';
 
 @injectable()
 export class PluginWorker {
 
     private worker: Worker;
+
     public readonly rpc: RPCProtocol;
+
     constructor() {
         const emitter = new Emitter<string>();
-        this.worker = new (require('../../hosted/browser/worker/worker-main'));
-        this.worker.onmessage = message => {
-            emitter.fire(message.data);
-        };
-        this.worker.onerror = e => console.error(e);
+
+        if (this.isElectron()) {
```

Comment:
I confirmed that the behavior works for the `browser` use-case, but if a user adds a theia frontend plugin to an electron use-case all other plugins fail to start (ex: `typescript-language-features`).

Reply:
That is a well-known and long-standing problem: https://github.com/eclipse-theia/theia/issues/6601

Reply:
If we want to disable loading plugins in the electron case (which is better than running into errors, I agree), we should do it in hosted-plugin.ts#startPlugins() instead of constructing a rpc endoint that doesn't do anything.

Reply:
> That is a well-known and long-standing problem: #6601

@tsmaeder I know about #6601, the problem I wanted to highlight wasn't about theia frontend plugins but the fact that if a a theia frontend plugin is added in the electron application it will cause vscode plugins such as the typescript-language-features to not start. I was hopping for a more elegant way to handle the frontend theia plugins in electron as to not fail other valid plugins.

Reply:
Ah, so https://github.com/eclipse-theia/theia/pull/9715#discussion_r669315368 is what you want, right?

Reply:
Right, if it means not failing other valid plugins I'm all for it :)

Reply:
Reworked